Upgrade to blacklight 7.22

sandbergja opened the pull request:
After this is merged:
- Start work on the countless deprecation warnings
- Evaluate if we still need the blacklight_advanced_search gem, or if the [advanced search functionality in core blacklight](https://github.com/projectblacklight/blacklight/pull/2363) meets our needs.
- Migrate our facets to use the ViewComponent versions.  The classification facet is somewhat migrated, the rest are using mainly partials with a bit of components.

carolyncole reviewed (changes requested):
@sandbergja Great work!  Wow you really had to dig in to find some of these solutions!  I had a couple places I did not entirely understand the change.

sandbergja commented on spec/system/searching_spec.rb:
```diff
@@ -33,7 +33,7 @@
     it 'On-site label is green' do
       visit '/?f%5Baccess_facet%5D%5B%5D=In+the+Library&q=id%3Adsp*&search_field=all_fields'
-      expect(page).to have_selector '#documents > article.document.blacklight-senior-thesis.document-position-0 > div > div.record-wrapper > ul > li.blacklight-holdings > ul > li:nth-child(1) > span.availability-icon.badge.badge-success'
+      expect(page).to have_selector '#documents > article.document.blacklight-senior-thesis.document-position-1 > div > div.record-wrapper > ul > li.blacklight-holdings > ul > li:nth-child(1) > span.availability-icon.badge.badge-success'
```
No change in order; Blacklight just changed these to start at 1 instead of 0.

sandbergja commented:
Thanks for the review, @carolyncole.  I rebased this and fixed a few of the things you caught.  I still need to look into the facets_for_advanced_search override and the endless constraints upon constraints when using the pivot facet.

I also caught a regression on the search results page in the mobile view of pivot facets:

On production:
<img width="307" alt="Screen Shot 2022-02-18 at 2 08 50 PM" src="https://user-images.githubusercontent.com/7086943/154768330-b2445633-c5b0-4233-a842-bb300893cbf8.png">

On qa, with this branch deployed:
<img width="308" alt="Screen Shot 2022-02-18 at 2 08 34 PM" src="https://user-images.githubusercontent.com/7086943/154768358-94cba9d5-3eaa-49c9-b00d-1b4ee2a864bc.png">

sandbergja commented on app/helpers/catalog_helper.rb:
```diff
@@ -53,7 +53,7 @@ def render_search_to_page_title(params)
 
   def render_top_field?(document, field_name)
-    presenter(document).fields_to_render[field_name].nil? && document[field_name].present? &&
+    !presenter(document).fields_to_render.map { |item| item }.include?(field_name) && document[field_name].present? &&
```
I reverted this change back to how it was.

sandbergja replied:
Oof, reverting was a bad move -- it caused over 70 failing specs!  I put your version back, just without the map.

sandbergja replied:
Okay, I finally get why the map is necessary.  `fields_to_render` is a pretty weird situation that only gives us what we need if we pass a block to it: https://github.com/projectblacklight/blacklight/blob/44e7ff915af51ce472a4403bb198cc16c77bf40c/app/presenters/blacklight/document_presenter.rb#L22.  So I am leaving it with how you had it, @carolyncole.

sandbergja commented on app/models/search_builder.rb:
```diff
@@ -51,6 +52,11 @@ def run_advanced_parse?(solr_parameters)
       cleaned_query(solr_parameters[:q]) == solr_parameters[:q]
   end
 
+  def facets_for_advanced_search_form(solr_p)
```
No, this is just for the solr query used to generate all the facet values and counts for the advanced search screen.  I'm really glad you asked, though, especially since the version I had earlier got all the wrong counts -- here's an updated version.
